pkg/abi/linux/netdevice.go

```diff
@@ -61,6 +61,7 @@ func (ifr *IFReq) SetName(name string) {
 	for i := n; i < len(ifr.IFName); i++ {
 		ifr.IFName[i] = 0
 	}
+	ifr.IFName[len(ifr.IFName) - 1] = 0
```

Comment:
I wonder if line 60 shouldn't be changed to this instead? That way we don't copy over the last byte in the first place?
n := copy(ifr.IFName[:len(if.IFName)-1], []byte(name))

Also, add a comment somewhere that indicates that we need to preserve the null terminator at the end of the name.

Reply:
Also a simple table-driven test that ensures this is working as expected in a few different scenarios would be nice. The nominal one being when len(name) exceeds len(if.IFName)-1

Reply:
I am not sure about your suggestion with line 60, as it is usually helpful for the readers when this mitigation is as explicit as possible.
I can do both, in order to be on the safe side.

Let me know if you disagree :)

Reply:
I think updating line 60 and adding a comment is fine. It conveys the idea that only the 0:len-1 elements are writable in the slice.

Even doing it afterward, it's not immediately clear why it's necessary on the surface as it appears to be duplicating work (probably why it was omitted in the first place).